application/src/main/java/org/opentripplanner/updater/vehicle_rental/datasources/gbfs/GbfsFeedLoaderImpl.java

```diff
@@ -135,24 +164,64 @@ private T getData() {
     }
 
     private boolean fetchData() {
-      T newData = fetchFeed(url, httpHeaders, otpHttpClient, implementingClass);
-      if (newData == null) {
-        LOG.warn("Could not fetch GBFS data for {}. Retrying.", url);
+      try {
+        GBFSFeedResponse<T> response = fetchFeedWithResponse(
+          url,
+          httpHeaders,
+          etag,
+          otpHttpClient,
+          otpHttpResponse ->
+            new GBFSFeedResponse<>(
+              otpHttpResponse.statusCode() == 200
+                ? objectMapper.readValue(otpHttpResponse.body(), implementingClass)
```

Comment:
JSON exceptions are not caught. See: https://github.com/opentripplanner/OpenTripPlanner/blob/cfe348a468736325c21a6364ffffd9b0abee61b1/application/src/main/java/org/opentripplanner/framework/io/OtpHttpClient.java#L129